src/components/general/Popup.js

```diff
@@ -25,68 +85,102 @@ const Popup = ({
+      let top
       if (!triggerRef) {
-        wrapper.current.style.left = `${
-          rect.x - wrapperRect.width / 2 + rect.width / 2
-        }px`
-        wrapper.current.style.top = `${rect.y}px`
+        left = rect.x - wrapperRect.width / 2 + rect.width / 2
+        top = rect.y
       } else {
-        switch (position) {
+        const triggerRect = triggerRef.current.getBoundingClientRect()
+        switch (computedPosition) {
           case 'topCenter':
-            wrapper.current.style.top = `${
-              rect.y - wrapperRect.height - triggerRect.height
-            }px`
-            wrapper.current.style.left = `${
-              rect.x - wrapperRect.width / 2 + triggerRect.width / 2
-            }px`
+            top = rect.y - wrapperRect.height - triggerRect.height
+            left = rect.x - wrapperRect.width / 2 + triggerRect.width / 2
             break
           case 'topRight':
-            wrapper.current.style.top = `${
-              rect.y - wrapperRect.height - triggerRect.height
-            }px`
-            wrapper.current.style.left = `${rect.x}px`
+            top = rect.y - wrapperRect.height - triggerRect.height
+            left = rect.x
             break
           case 'topLeft':
-            wrapper.current.style.top = `${
-              rect.y - wrapperRect.height - triggerRect.height
-            }px`
-            wrapper.current.style.left = `${
-              rect.x - wrapperRect.width + triggerRect.width
-            }px`
+            top = rect.y - wrapperRect.height - triggerRect.height
+            left = rect.x - wrapperRect.width + triggerRect.width
             break
           case 'bottomRight':
-            wrapper.current.style.top = `${rect.y}px`
-            wrapper.current.style.left = `${rect.x}px`
+            top = rect.y
+            left = rect.x
             break
           case 'bottomLeft':
-            wrapper.current.style.top = `${rect.y}px`
-            wrapper.current.style.left = `${
-              rect.x - wrapperRect.width + triggerRect.width
-            }px`
+            top = rect.y
+            left = rect.x - wrapperRect.width + triggerRect.width
             break
           case 'bottomCenter':
           default:
-            wrapper.current.style.top = `${rect.y}px`
-            wrapper.current.style.left = `${
-              rect.x - wrapperRect.width / 2 + triggerRect.width / 2
-            }px`
+            top = rect.y
+            left = rect.x - wrapperRect.width / 2 + triggerRect.width / 2
         }
       }
+      wrapper.current.style.top = `${top}px`
+      wrapper.current.style.left = `${left}px`
+
+      setOriginalTopPosition(top)
+    }
+  }, [
+    isVisible,
+    wrapper,
+    placeholder,
+    triggerRef,
+    computedPosition,
+    setOriginalTopPosition
+  ])
+
+  // Callback invoked on each scroll. Uses original top position to scroll with content.
+  const watchScroll = useCallback(
+    (scrollParent, initialScrollPosition) => {
```

Comment:
nbd, but I think this function can just take the scrollParent as argument, given that initialScrollPosition is just the scrollParent.scrollTop?

Reply:
it's not actually -- initial position refers to the position when the popup first opened, whereas scrollParent.scrollTop gets the "current" scroll pos. Adding a comment